README.md

```diff
@@ -1,8 +1,23 @@
-# Babylon Finality Gadget
+# ⚠️ NOT FOR PRODUCTION USE ⚠️
 
-The Babylon Finality Gadget is a program that can be run by users of OP stack L2s to track consecutive L2 block quorum and query the BTC-finalised status of blocks.
+> **This repository is a Proof-of-Concept (POC) and example implementation 
+> to showcase how a Finality Gadget can work. It is not a production-ready 
+> system and should not be used in production environments. We do not advise 
+> companies or projects to use this codebase as-is.**
+
+# Rollup BSN Finality Gadget
+
+The **Rollup BSN Finality Gadget** is an off-chain program that can be 
+run by the Rollup BSN network or by users of Rollup BSN. Its primary 
+purpose is to track and determine whether a given L2 block is BTC-finalized. 
+A block is considered BTC-finalized if it has received signatures from a quorum 
+of registered Finality Providers (FPs) for that block, according to the rules and 
```

Comment:
There is another rule that a BTC staking finalised block at height `h` need to satisfy: its previous block `h-X` has to also be BTC staking finalised as well, where `X` is the finality signature interval.

Reply:
Fixed 0c0de49d50bc2c18cb3a9ce9b11a82b67436d010